lib/mobility/attributes.rb

```diff
@@ -116,52 +116,16 @@ def self.plugin(name)
     # @return [Array<String>] Array of names
     attr_reader :names
 
-    # Backend class
-    # @return [Class] Backend class
-    attr_reader :backend_class
-
-    # Name of backend
-    # @return [Symbol,Class] Name of backend, or backend class
-    attr_reader :backend_name
-
     # @param [Array<String>] attribute_names Names of attributes to define backend for
-    # @param [Symbol] method One of: [reader, writer, accessor]
-    # @param [Symbol,Class] backend Backend to use
     # @param [Hash] backend_options Backend options hash
-    # @raise [ArgumentError] if method is not reader, writer or accessor
-    def initialize(*attribute_names, method: :accessor, backend: Mobility.default_backend, **backend_options)
-      raise ArgumentError, "method must be one of: reader, writer, accessor" unless %i[reader writer accessor].include?(method)
+    def initialize(*attribute_names, **backend_options)
       @options = Mobility.default_options.to_h.merge(backend_options)
       @names = attribute_names.map(&:to_s).freeze
-
-      @backend_name = backend
-
-      unless backend.nil?
-        attribute_names.each do |name|
-          define_backend(name)
-          define_reader(name) if %i[accessor reader].include?(method)
-          define_writer(name) if %i[accessor writer].include?(method)
-        end
-      end
     end
 
-    # Setup backend class, include modules into model class, include/extend
-    # shared modules and setup model with backend setup block (see
-    # {Mobility::Backend::Setup#setup_model}).
-    # @param klass [Class] Class of model
     def included(klass)
-      if backend_name
-        @backend_class = Backends.load_backend(backend_name)
-          .for(klass)
-          .with_options(@options.merge(model_class: klass))
-
-        klass.include InstanceMethods
-        klass.extend ClassMethods
-
-        backend_class.setup_model(klass, names)
-
-        backend_class
-      end
+      klass.extend ClassMethods
+      nil
```

Comment:
Note to self: `nil` is bad here, this is a placeholder for something better. :thinking: